blockchain/src/pretty.rs

```diff
@@ -0,0 +1,405 @@
+use std;
+use std::fmt;
+use std::string::String;
+
+use ansi_term::Colour;
+
+use block;
+use genesis;
+use normal;
+use types;
+use wallet_crypto;
+
+static DISPLAY_INDENT_SIZE: usize = 4; // spaces
+static DISPLAY_INDENT_LEVEL: usize = 0; // beginning starts at zero
+static DISPLAY_USE_COLOR: bool = false; // no color for display implementations
+
+type Assoc = (String, Val);
+
+// core pretty-printer type which facilitates homogenous ordered key-value pairs
+pub enum Val {
+    Single(Box<fmt::Display>),
+    Pairs(Option<Colour>, Vec<Assoc>),
+}
+
+// XXX: this class is related to "into" somehow
```

Comment:
It does, formally, but I don't think this is particularly useful a remark. `Into` is used as a generic conversion mechanism in macros. But you probably don't want to convert implicitly to a `Pretty` anyway. Plus it is recommended to not implement `Into` directly, but to implement `From` instead. Which is impractical for this use-case.

Reply:
thanks for explaining!